src/core/border_router/routing_manager.cpp

```diff
@@ -716,13 +716,13 @@ void RoutingManager::SendRouterAdvertisement(RouterAdvTxMode aRaTxMode)
     // RA message max length is derived to accommodate:
     //
     // - The RA header,
-    // - At most one PIO (for local on-link prefix),
+    // - At most two PIOs (for current and old local on-link prefixes),
```

Comment:
If we switch to a new network again during this deprecation period, there can two deprecated PIOs in this case. To handle it, we will want to keep a set of deprecated prefixes (e.g. `mDeprecatedOnLinkPrefixes`) and have a `MAX_DEPRECATED_PREFIXES_NUM` to limit the size of the set. Our current implementation in this PR is setting `MAX_DEPRECATED_PREFIXES_NUM` to `1` and I think it's okay since changing the network twice during the deprecation period is very likely a corner case which won't happen in production.

Reply:
Yes. I was considering the same model (extending it to an array of deprecating prefixes). I was thinking to add this in a future/follow-up PR when we add mechanism to save the old prefixes in non-volatile settings (to deprecate them on reboot/reset).

Reply:
>  I was thinking to add this in a future/follow-up PR when we add mechanism to save the old prefixes in non-volatile settings (to deprecate them on reboot/reset)

Thanks, this SGTM!

Reply:
One more comment, for saving old prefixes in non-volatile settings, I think it's necessary to save discovered on-link prefixes as well (considering the situation where the active on-link prefix is generated by BR's in another Thread network).